app/src/main/java/com/sdp13epfl2021/projmag/adapter/ProjectAdapter.kt

```diff
@@ -40,8 +40,18 @@ class ProjectAdapter(
     var datasetAll: List<ImmutableProject> = emptyList()
     val dataset: MutableList<ImmutableProject> = datasetAll.toMutableList()
 
+    /**
+     * Sort the list of projects, first are the projects not taken, sorted with newest date first.
+     * If the activity is created with a link, it will put it at the top.
+     */
     private fun sortDataset() {
```

Comment:
I think a better name for this method is sortProjects

Reply:
Yes, I didn't change the name because someone else is maybe still refactoring the adapter package ?
And others things should also be refactor such as dataset/datasetAll/etc.